server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2188,7 +2188,13 @@ public UserVm updateVirtualMachine(UpdateVMCmd cmd) throws ResourceUnavailableEx
         }
 
         if (details != null && !details.isEmpty()) {
-            vmInstance.setDetails(details);
+            _vmDao.loadDetails(vmInstance);
```

Comment:
I think all we need is here, the side effect is that now users won't be able to remove any details; only add/update details.